### ISSUE TITLE
Hunter pets lose all Buffs/Debuffs when called upon

### ISSUE THREAD

**Issue by knasen88** [elysium-project/server#470]
All pets currently lose all their positive and negative status effects upon being spawned. This means you can't bring any buffs on your pets when jumping over a cliff in raids. neither Dismiss or outranging works.

Source: https://youtu.be/ckrcaDuizs4?t=11m16s 

his pet get sheeped, he then runs away and calls pet and it retains the effect. 11m36~

**Comment by lequintal69** [comment]
I hope its true. I thought it wasnt the case since hakkar plague event (pet getting debuff, dismissed and resummoned in capitals), but apparently they just made pets immuned to corrupted blood

**Comment by sknnbones** [comment]
I think its an intended non-blizzlike fix for old school "griefing"

People used to dismiss with Living Bomb and other AoE debuffs, and that would freeze the countdown timer, allowing people to summon pets with these debuffs in lowbie areas instantly killing everyone nearby.

**Comment by knasen88** [comment]
That's stupid fix, pets are big part of hunter damage and not letting us use them with world buffs is bullshit. I hope this has just been overlooked and will be corrected.

**Comment by Phibs77** [comment]
This still doesn't work and cripples hunter.

**Comment by BrynjaDarrow** [comment]
I think #1855 has something to do with it.
It seems like everytime your pet is reloaded, a new instance of your pet is created, so it doesn't keep the buffs since its not the "same" pet.

**Comment by knasen88** [comment]
I can also mention a server crash wipes all pet buffs

**Comment by Voff89** [comment]
With the fix to the scrip dismissal pet macro, this needs some serious attention, not being able to dismiss pets without losing world buffs is a joke.

**Comment by fitigf15** [comment]
Hello, I personally didn't know about that pet dismiss instant macro back in the days, but when I started playing again in private servers I found it casually by trying to make an "all-in-one" macro for things like call, feed, dismiss, resurrect in one button.

The thing is that I can't prove that it was doable in retail, but I've been using it in every single private server with completely different cores. This script was working on Elysium the same way it was working on Feenix, Retro-Wow, Kronos, Nostalrius and probably all the other vanilla private servers that have different core and scripting, thats why I can assume that it worked on retail aswell.

I understand that instant dismiss pet got disabled, since it was super exploitable, but most of the hardcore players were using it wisely, ONLY in PVE and ONLY to preserve world buffs, since it was the only way to do it. If pet cannot keep buffs, that means that hunter has to ask the entire raid group to rebuff every time the pet gets dismissed, also to blow up jujus and / or scrolls every dismiss. That also means a huge loss of time, consumables, gold, and also also pissing off all the paladins, druids, priests, even warriors by asking for a rebuff every single time.

In summary, if pets cannot keep buffs at all, while hunters depend A LOT on their pets to deal damage, this change will make hunter class underperform even more than before this changes, since  the hunter class has by default less DPS tools than the other DPS classes.

If I had to vote for something, I'd vote for having the 5 second cast dismiss pet, but keeping the buffs/debuffs as back in the days, also when pet auto dismisses because it's too far away from the hunter, should keep the buffs/debuffs (as far as I can recall).

Cadet - Night Elf Hunter - <Progress> - Elysium

**Comment by lequintal69** [comment]
"most of the hardcore players were using it wisely, ONLY in PVE and ONLY to preserve world buffs"

lol

**Comment by sadfx** [comment]
what you think of as "hardcore pve" is casual

**Comment by knasen88** [comment]
still not fixed REOPEN

**Comment by yankellch** [comment]
Damage of hunter's pet has been nerffed so much here in this project,as the only thing that hunter can depend on to deal in DPS，pet should exist with casted buffs in battles which is more fair for this class in vanilla wow.

**Comment by BrynjaDarrow** [comment]
It has been fixed on Anathema tho afaik, the patch just isn't live yet on Darrow/Elysium.
